### ISSUE TITLE
App crashes on opening notifications

### ISSUE THREAD

**Issue by qaysdwekat** [smooch/smooch-ios#192]
The app crashes on opening a push notification while the app is killed.
After the chat view fully loaded the app crashed.

Here's the crash report:
```
Thread 0 name:  Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   UIKit                         	0x2556ba56 -[UITableViewRowData numberOfRowsInSection:] + 54
1   Smooch                        	0x03e379d0 0x3dcb000 + 444880
2   Smooch                        	0x03dfe256 0x3dcb000 + 209494
3   Smooch                        	0x03e37910 0x3dcb000 + 444688
4   Smooch                        	0x03e37456 0x3dcb000 + 443478
5   CoreFoundation                	0x20e6d734 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
6   CoreFoundation                	0x20e6d13e _CFXRegistrationPost + 390
7   CoreFoundation                	0x20e6cf1c ___CFXNotificationPost_block_invoke + 40
8   CoreFoundation                	0x20ec3c6a -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1334
9   CoreFoundation                	0x20dcd082 _CFXNotificationPost + 486
10  Foundation                    	0x2160e5de -[NSNotificationCenter postNotificationName:object:userInfo:] + 74
11  Smooch                        	0x03e2ecec 0x3dcb000 + 408812
12  libdispatch.dylib             	0x20a393ea _dispatch_barrier_sync_f_slow_invoke + 494
13  libdispatch.dylib             	0x20a2980e _dispatch_client_callout + 22
14  libdispatch.dylib             	0x20a37ba8 _dispatch_main_queue_callback_4CF$VARIANT$mp + 1524
15  CoreFoundation                	0x20e7db6c __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 8
16  CoreFoundation                	0x20e7c066 __CFRunLoopRun + 1574
17  CoreFoundation                	0x20dcb228 CFRunLoopRunSpecific + 520
18  CoreFoundation                	0x20dcb014 CFRunLoopRunInMode + 108
19  GraphicsServices              	0x223bbac8 GSEventRunModal + 160
20  UIKit                         	0x2549f188 UIApplicationMain + 144
21  Yamsafer                      	0x0009a254 main + 143956 (AppDelegate.swift:20)
22  libdyld.dylib                 	0x20a73872 start + 2
```

Smooch version: 6.7.1
Device: iPad2,2 - iOS 9.3.5

Thanks for helping!

**Comment by mspensieri** [comment]
Thanks for the report. Do you have the details of the exception that was thrown? Your report includes the call stack but not the actual error message

**Comment by qaysdwekat** [comment]
This's the full details of the crash:
[Yamsafer  3-20-18, 6-25 PM.txt](https://github.com/smooch/smooch-ios/files/1830531/Yamsafer.3-20-18.6-25.PM.txt)

**Comment by mspensieri** [comment]
Thank you. Some follow-up questions:

1. Does this only reproduce on iOS 9.x? Or is 10+ affected too?
2. You say "while the app is killed". This means you can correctly open the app from a push notification if the app is still alive, but in the background?
3. Is it related to the conversation state in some way? i.e. if you delete the messages in the convo, does the crash still reproduce?
4. If the conversation state matters, can you give a brief description of what the history looks like? Any rich message types?

**Comment by mspensieri** [comment]
An attempt was made to fix this in [v6.7.2](https://github.com/smooch/smooch-ios/releases/tag/6.7.2). Please test and let us know if the issue persists

**Comment by mspensieri** [comment]
Closing due to inactivity. Please reopen if the issue occurs with the latest version
